core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -237,6 +246,43 @@ class BlockManagerMasterEndpoint(
     blockManagerIdByExecutor.get(execId).foreach(removeBlockManager)
   }
 
+  private def replicateOneBlock(
+      execId: String,
+      blockId: BlockId,
+      excludeExecutors: Seq[String],
+      context: RpcCallContext): Unit = {
+    logDebug(s"replicating block $blockId")
+    val excluded = excludeExecutors.flatMap(blockManagerIdByExecutor.get)
+    val response: Option[Future[Boolean]] = for {
+      blockManagerId <- blockManagerIdByExecutor.get(execId)
+      info <- blockManagerInfo.get(blockManagerId)
+      replicaSet <- blockLocations.asScala.get(blockId)
+      replicas = replicaSet.toSeq
+      maxReps = replicaSet.size + 2
```

Comment:
Why `+ 2`? Good to add comment explaining.

Reply:
commented